error_functions.py

```diff
@@ -0,0 +1,26 @@
+from cmath import log
+from typing import Callable, Optional
+
+ErrorFunctionType = Callable[[float, float, Optional[bool]], float]
+
+class ErrorFunctions:
+    @staticmethod
+    def cross_entropy(calculated: float, expected: float, derivative: bool = False) -> float:
+        if derivative:
+            return expected / calculated - (1 - expected) / (1 - calculated)
+        else:
+            return -expected * log(calculated) - (1 - expected) * log(1 - calculated)
+
+    @staticmethod
+    def mean_squared(calculated: float, expected: float, derivative: bool = False) -> float:
+        if derivative:
+            return expected - calculated
```

Comment:
Are you sure about derivative here? I would expect it should rather be something similar to described here: https://towardsdatascience.com/gradient-descent-from-scratch-e8b75fa986cc

Reply:
I think so: if out function is `f(x,y) = 1/2 * (x-y)^2 = 1/2 (x^2 - 2xy + y^2)`, then derivative `(df(x,y)/x) = 1/2 * (2x - 2y) = x - y` and the same for `(df(x,y)/dy)`

Reply:
Yep, you are right :)